src/createProjector.ts

```diff
@@ -128,7 +141,19 @@ const createProjector: ProjectorFactory = createWidgetBase
 					};
 				});
 
-				projectorData.projector.append(projectorData.root, render);
+				switch (type) {
+					case 'append':
```

Comment:
On second thought, it might be nice if we just make the types into functions, so rather than having `attach()`, we just have `append()`, `merge()` and `replace()`. That way we get intellisense and the end user can't use an incorrect attach type?